cpp/cmake/StandardConfig.cmake

```diff
@@ -209,7 +209,7 @@ function(StandardConfig config_type)
       set(ENABLE_CHECK_FORMAT OFF)
     else(NOT EXISTS "${FORMAT_CMAKE_PATH}")
       set(FORMAT_SKIP_CMAKE YES CACHE BOOL "" FORCE)
-      add_subdirectory("${FORMAT_CMAKE_PATH}" EXCLUDE_FROM_ALL)
+      add_subdirectory("${FORMAT_CMAKE_PATH}" "${FORMAT_CMAKE_PATH}" EXCLUDE_FROM_ALL)
```

Comment:
> If binary_dir is not specified, the value of source_dir, before expanding any relative path, will be used (the typical usage)

FORMAT_CMAKE_PATH is likely an absolute path, making it output to `FORMAT_CMAKE_PATH` directly, which is wrong (though it should not actually output). Your change feels wrong to me. What's the exact issue you're facing with conan why it's needed?

Reply:
Thanks, I think it's due to different versions of CMake we have. Here it specifically complaints `FORMAT_CMAKE_PATH` is outside source dir, i.e. it's absolute path. Use `${CMAKE_SOURCE_DIRECTORY}/format_cmake` then, not `${FORMAT_CMAKE_PATH}`